build_tools/aws-sdk-code-generator/templates/endpoints_plugin.mustache

```diff
@@ -86,12 +86,8 @@ The endpoint provider used to resolve endpoints. Any object that responds to
         end
 
         def parameters_for_operation(context)
-          case context.operation_name
-          {{#endpoint_classes}}
-          when :{{operation_name}}
-            {{module_name}}::Endpoints::{{class_name}}.build(context)
-          {{/endpoint_classes}}
-          end
+          operation_class = {{module_name}}::Endpoints::OPERATION_PARAM_MAP[context.operation_name]
+          operation_class ? operation_class.build(context) : {{module_name}}::EndpointParameters.create(context.config)
```

Comment:
Parameters for operation should probably fail if the operation doesn't exist, maybe an argument or runtime error is appropriate. If the default was somehow used, that could be difficult to track down.

Regardless, if we must keep this, please don't make it a ternary.

Reply:
Parameter classes now only exist when the operation needs specific configuration (which reduces size a ton since most operations dont) so I think we still need this.

Removed the ternary.

Reply:
Oh I see. So this isn't every operation, just ones that don't have config based ones?

Reply:
Maybe the map name could be different to indicate that it's not all the operations.

Reply:
yeah exactly. Changed to OPERATION_CUSTOM_PARAMS_MAP

Reply:
Moved method to endpoints module and use case statement.